presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java

```diff
@@ -111,6 +111,12 @@
     @GuardedBy("this")
     private boolean verboseExceededMemoryLimitErrorsEnabled;
 
+    @GuardedBy("this")
+    private boolean heapDumpOnExceededMemoryLimitEnabled;
+
+    @GuardedBy("this")
+    private String heapDumpFilePath;
```

Comment:
I'm fine with this being `null` given the specialized nature of the whole thing, but if someone picky comes along to commit they might want this to be Optional.